src/Tests/WebApp.fs

```diff
@@ -186,4 +186,43 @@ let tests = testList "Web App Tests" [
         Expect.sequenceEqual (wa.Identity.UserAssignedIdentities |> Seq.map(fun r -> r.Key)) [ "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', 'test2')]"; "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', 'test')]" ] "Should have two user assigned identities"
 
     }
+
+    test "Handles use_extension correctly" {
+        let wa = webApp { name "xyz"; use_extension "abc"; }
+        let resources = wa |> getResources
+        let sx = resources |> getResource<SiteExtension> |> List.head
+        let r  = sx :> IArmResource
+
+        Expect.equal sx.SiteName (ResourceName "xyz") "Extension knows the site name"
+        Expect.notEqual sx.Location (Location "") "Location has a default value"
+        Expect.equal sx.Name (ResourceName "abc") "Extension name is correct"
+        Expect.equal r.ResourceName (ResourceName "xyz/abc") "Resource name composed of site name and extension name"
+    }
+
+    test "Handles multiple use_extension correctly" {
+        let wa = webApp { name "xyz"; use_extension "abc"; use_extension "def" }
+        let resources = wa |> getResources |> getResource<SiteExtension>
+
+        Expect.equal resources.Length 2 "Both extensions captured"
+
+        // We will make no assertion on the ordering of sx1 and sx2
```

Comment:
Consider using `Expect.sequenceEqual` here.

Reply:
OK, looks a lot better now. Took a different approach.
I also want to make this change (define `testLocation`)  so that I don't have to hard-code the location in my test, and can just refer to `testLocation`
```
let testLocation = Location.WestEurope

let tests = testList "Web App Tests" [
    let getResources (wa:WebAppConfig) = (wa :> IBuilder).BuildResources testLocation
```

Reply:
Then my test would look like this
```
    test "Handles multiple use_extension correctly" {
        let wa = webApp { name "siteX"; use_extension "abc"; use_extension "def" }
        let resources = wa |> getResources |> getResource<SiteExtension>

        Expect.sequenceEqual
            (List.sortBy (fun (x : SiteExtension) -> x.Name) resources)
            [
                { Location = testLocation; Name = ResourceName "abc"; SiteName = ResourceName "siteX" }
                { Location = testLocation; Name = ResourceName "def"; SiteName = ResourceName "siteX" }
            ]
            ""
    }
```